### PR TITLE
Added MVG errors with better error messages

### LINKED ISSUE #43: Add more descriptive exceptions to MVG
The errors is probably the weakest point of MVG right now.

We should probably implement our own exceptions and improve the error messages. To do this we might have to make some changes to the server side to send additional information about the type of error that is sent with the error response

### PR DESCRIPTION
**Pull request by OskarLiew**
# Description

Added new MVG specific errors. In particular `MVGAPIError` that will automatically read the details of the error message

New dependencies: (Added to requirements_*.txt)

Fixes #43 (potential issues that are fixed by this PR)

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue -> bump patch)
- [x] New feature (non-breaking change which adds functionality -> bump minor version)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected -> bump major version)
- [ ] Documentation Update

## Checklist

- [x] I have added tests that prove that my fix/feature works
- [x] Linters pass locally and I have followed PEP8 code style
- [x] New and existing tests pass locally
- [x] I have updated the documentation if needed
- [x] I have commented hard-to-understand areas in the code

## Requirements

- [x] I have updated the MVG version

### CODE REVIEW

File: tests/test_errors.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            raise requests.HTTPError()
+
+
+def mock_fail_request(json_body):
+    return MockResponse(400, "Bad Request", json_body)
+
+
+def assert_fail_response(msg, detail="No error message"):
+    assert msg == f"400 - Bad Request: {detail}"
+
```
Comment (tuix) [inline tests/test_errors.py:31]: How are these two functions used? Do not seem to be invoked by other functions, and won't be called as a test function by pytest either?

**Review by tuix — CHANGES_REQUESTED** [review]
beautifully implemented. Thanks for taking care of this. Just a small question about some functions

File: tests/test_errors.py (hunk 1 above)
Comment (OskarLiew) [reply tests/test_errors.py:31]: These were just something I was playing around with but decided was overkill. Seems I forgot to remove the functions

File: docs/source/content/examples/error_handling.ipynb [hunk 2]
```diff
@@ -157,7 +157,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "{\"api\": {\"name\": \"MultiViz Engine API\", \"version\": \"v0.2.7\", \"swagger\": \"http://api.beta.multiviz.com/docs\"}}\n"
+      "{\"api\": {\"name\": \"MultiViz Engine API\", \"version\": \"0.0.0-dev0\", \"swagger\": \"http://localhost:8000/docs\"}}\n"
```
Comment (vnadhan) [inline docs/source/content/examples/error_handling.ipynb:160]: localhost:8000 ?

File: docs/source/content/examples/sources_and_measurements.ipynb [hunk 3]
```diff
@@ -154,8 +154,8 @@
      "data": {
       "text/plain": [
        "{'api': {'name': 'MultiViz Engine API',\n",
-       "  'version': 'v0.2.7',\n",
-       "  'swagger': 'http://api.beta.multiviz.com/docs'}}"
+       "  'version': '0.0.0-dev0',\n",
+       "  'swagger': 'http://localhost:8000/docs'}}"
```
Comment (vnadhan) [inline docs/source/content/examples/sources_and_measurements.ipynb:158]: again localhost:8000 here ..

**Review by vnadhan — CHANGES_REQUESTED** [review]

**Comment by vnadhan** [comment]
Just found two minor issues in the documentation, otherwise good work @OskarLiew!

**Comment by vnadhan** [comment]
I tested this branch locally and it much easier to know what caused an error, however they are too lengthy with unwanted characters or detail. This is an error that happened when a source id has a space in it. 
image

**Review by vnadhan — DISMISSED** [review]

**Review by vnadhan — APPROVED** [review]

**Pull request merged**